src/approaches/grammar_search_invention_approach.py

```diff
@@ -273,9 +416,12 @@ def _get_current_predicates(self) -> Set[Predicate]:
     def learn_from_offline_dataset(self, dataset: Dataset) -> None:
         # Generate a candidate set of predicates.
         print("Generating candidate predicates...")
-        grammar = _create_grammar(CFG.grammar_search_grammar_name, dataset)
+        grammar = _create_grammar(CFG.grammar_search_grammar_name, dataset,
+                                  self._initial_predicates)
         candidates = grammar.generate(max_num=CFG.grammar_search_max_predicates)
-        print(f"Done: created {len(candidates)} candidates.")
+        print(f"Done: created {len(candidates)} candidates:")
+        for predicate in candidates:
```

Comment:
Won't this be a massive printout?

Reply:
right now it's very manageable because we're not proposing that many. it's been helpful for me but we can remove later if it's too much.